source/shared/core_util.cpp

```diff
@@ -485,8 +493,9 @@ namespace data_classification {
             convert_sensivity_field(stmt, encoding, idptr, idlen, (char**)&id);
             pair->id = id;
 
-            pairs.push_back(pair.get());
+            name_id_pair* a_pair = pair;
             pair.transferred();
+            pairs->push_back(a_pair);
```

Comment:
This seems to be the only place where converting pairs to a pointer has any effect, but it's not clear to me why this would solve the memory issues?